spikeextractors/extractors/neoextractors/neobaseextractor.py

```diff
@@ -0,0 +1,139 @@
+        # there is no garranty that ids/names are unique on some formats
+        raw_traces = self.neo_reader.get_analogsignal_chunk(block_index=self.clock_index, seg_index=self.seg_index,
+                                i_start=start_frame, i_stop=end_frame,
+                               channel_indexes=None, channel_names=None, channel_ids=channel_ids)
+        
+        # rescale gtrace to natural units
+        scaled_traces = self.neo_reader.rescale_signal_raw_to_float(raw_traces, dtype='float32',
+                                    channel_indexes=None, channel_names=None, channel_ids=channel_ids)
+        # and then to uV
+        scaled_traces *= self.additional_gain
+        
+        # fortunatly neo works with (samples, channels) strides
+        # so transpose to spieextractors wolrd
+        scaled_traces = scaled_traces.transpose()
+        
+        return scaled_traces
+        
+
+    def get_num_frames(self):
+        # channel_indexes=None means all channels
+        n = self.neo_reader.get_signal_size(self.block_index, self.seg_index, channel_indexes=None)
+        return n
+
+    def get_sampling_frequency(self):
+        # channel_indexes=None means all channels
+        sf = self.neo_reader.get_signal_sampling_rate(channel_indexes=None)
+        return sf
+
+    def get_channel_ids(self):
+        chan_ids = self.neo_reader.header['signal_channels']['id']
+        # in neo there is not garranty that chann ids are unique
+        # for instance Blacrock can have several times the same chan_id 
+        # different sampling rate
+        # so check it
+        assert np.unique(chan_ids).size == chan_ids.size, 'In this format channel ids are not unique'
+        # to avoid this limitation this could return chan_index which is 0...N-1
+        return chan_ids
+
+
+class NeoBaseSortingExtractor(SortingExtractor, _NeoBaseExtractor):
+    def __init__(self, **kargs):
+        _NeoBaseExtractor.__init__(self, **kargs)
+        
+        # the sampling frequency is quite tricky because in neo
+        # spike are handle in s or ms
+        # internally many format do have have the spike time stamps
+        # at the same speed as the signal but at a higher clocks speed.
+        # here in spikeinterface we need spike index to be at the same speed 
+        # that signal it do not make sens to have spikes at 50kHz sample
+        # when the sig is 10kHz.
+        # neo handle this but not spikeextractors
+        
+        self._handle_sampling_frequency()
+        #self._sampling_frequency = None
+    
+    def _handle_sampling_frequency(self):
+        # must handle :
+        #   _sampling_frequency
+        #   _time_stamp0
+        raise(NotImplementedError)
+    
+    def get_unit_ids(self):
+        # should be this
+        unit_ids = self.neo_reader.header['unit_channels']['id']
+        # in neo unit_ids are string so here we take index
+        unit_ids = np.arange(unit_ids.size, dtype='int64')
```

Comment:
shouldn't we be recasting these strings as ints? I don't think this will correctly handle unit_ids that are not  0-indexed. I take it we shouldn't be making this assumption. Otherwise we wouldn't have `get_unit_ids` at all, we'd just pass the number of units.

Reply:
No in neo some ids for units can be something like
ch13#u89
so string anyway.

Reply:
This reflect how there units id are store inside the format. in spikeextractor it needs to be int and so np.arange.